src/Configuration.php

```diff
@@ -212,21 +212,47 @@ public function getAutoCommit(): bool
     }
 
     /**
-     * @param Middleware[] $middlewares
+     * @deprecated setMiddlewares is deprecated since doctrine/dbal 3.6. Use setMiddleware instead.
+     *
+     * @param Middleware[] $middleware
+     *
+     * @return $this
+     */
+    public function setMiddlewares(array $middleware): self
+    {
+        trigger_error('setMiddlewares is deprecated since doctrine/dbal 3.6. Use setMiddleware instead.', E_USER_DEPRECATED);
```

Comment:
DBAL 3.6 has been released already. We cannot introduce new deprecation in that version. All deprecation have to be moved to 3.7.x.

Reply:
Oh yes my bad. Thank you :)